R/tm_g_distribution.R

```diff
@@ -343,9 +336,84 @@ srv_distribution <- function(id,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
-    data_extract <- list(dist_i = dist_var, strata_i = strata_var, group_i = group_var)
 
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+    dists1 <- c("Fligner-Killeen",
+                "t-test (two-samples, not paired)",
+                "F-test",
+                "Kolmogorov-Smirnov (two-samples)",
+                "one-way ANOVA")
+    rule_req <- function(value) {
+        if (!shinyvalidate::input_provided(value))
+          "Please select stratify variable."
+    }
+    rule_dupl <- function(...) {
+          strata <- selector_list()$strata_i()$select
+          group <- selector_list()$group_i()$select
+          if (isTRUE(strata == group))
+            "Please select different variables for strata and group."
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = list(
+        dist_i = dist_var,
+        strata_i = strata_var,
+        group_i = group_var
+      ),
+      data,
+      select_validation_rule = list(
+        dist_i = shinyvalidate::sv_required("Please select a variable")
+      ),
+      filter_validation_rule = list(
+        strata_i = shinyvalidate::compose_rules(
+          crule(rule_req, isTRUE(input$dist_tests %in% dists1)),
+          crule(rule_dupl, identical(input$dist_tests, "Fligner-Killeen"))
```

Comment:
this is also applicable for the test part only and not the plot, check it on main.

Reply:
be aware of the tests applied to the distinct parts: displot, qqplot and tests